File: C#/백준/Silver/1463. 1로 만들기/1로 만들기.cs
using System;

namespace q{
    class w{
        static void Main(){
            int a = int.Parse(Console.ReadLine());
            int[] aa = {0,0,1,1,2,3,2};
            if(a<=6){
                Console.WriteLine(aa[a]);
            }
            else{
                int[] bb = new int[a+1];
                for (int i=0; i<7; i++){
                    bb[i] = aa[i];
                }
                for (int i=7; i<a+1; i++){
                    int[] cc = {1000000,1000000,1000000};
                    cc[0] = bb[i-1]+1;
                    if(i%2==0){
                        cc[1] = bb[i/2]+1;
                    }
                    if(i%3==0){
                        cc[2] = bb[i/3]+1;
                    }
                    bb[i] = cc.Min();
                }
                Console.WriteLine(bb[a]);
            }
        }
    }
}

File: Python/백준/Silver/1463. 1로 만들기/1로 만들기.py
a=int(input())
aa = [0,0,1,1,2,3,2]
if a in aa:
    print(aa[a])
else:
    for i in range(7,a+1):
        bb=[]
        bb.append(aa[i-1]+1)
        if(i%2==0):
            bb.append(aa[i//2]+1)
        if(i%3==0):
            bb.append(aa[i//3]+1)
        aa.append(min(bb))
    print(aa[a])